spock-core/src/main/java/org/spockframework/compiler/ConditionRewriter.java

```diff
@@ -530,15 +583,53 @@ private Expression rewriteStaticMethodCondition(StaticMethodCallExpression condi
     // rewriting has produced N/A's that haven't been realized yet, so do that now
     args.add(realizeNas(ConstantExpression.FALSE));
     args.add(new ConstantExpression(explicit));
+    args.add(new ConstantExpression(lastVariableNum));
 
-    return rewriteToSpockRuntimeCall(resources.getAstNodeCache().SpockRuntime_VerifyMethodCondition, condition, message, args);
+    return surroundWithTryCatch(
+        condition,
+        message,
+        rewriteToSpockRuntimeCall(
+            resources.getAstNodeCache().SpockRuntime_VerifyMethodCondition,
+            condition,
+            message,
+            args));
   }
 
-  private Expression rewriteOtherCondition(Expression condition, Expression message) {
+  private Statement rewriteOtherCondition(Expression condition, Expression message) {
     Expression rewritten = message == null ? convert(condition) : condition;
 
-    return rewriteToSpockRuntimeCall(resources.getAstNodeCache().SpockRuntime_VerifyCondition,
+    final Expression executeAndVerify = rewriteToSpockRuntimeCall(resources.getAstNodeCache().SpockRuntime_VerifyCondition,
         condition, message, Collections.singletonList(rewritten));
+
+    return surroundWithTryCatch(condition, message, executeAndVerify);
+  }
+
+  private TryCatchStatement surroundWithTryCatch(Expression condition, Expression message, Expression executeAndVerify) {
+    final TryCatchStatement tryCatchStatement = new TryCatchStatement(
+        new ExpressionStatement(executeAndVerify),
+        new EmptyStatement()
+    );
+
+    tryCatchStatement.addCatch(
+        new CatchStatement(
+            new Parameter(new ClassNode(Throwable.class), "throwable"),
+            new ExpressionStatement(
+                AstUtil.createDirectMethodCall(
+                    new ClassExpression(resources.getAstNodeCache().SpockRuntime),
+                    resources.getAstNodeCache().SpockRuntime_ConditionFailedWithException,
+                    new ArgumentListExpression(Arrays.asList(
+                        message == null ? new VariableExpression("$spock_valueRecorder") : ConstantExpression.NULL, // recorder
```

Comment:
is there a reason why "$spock_valueRecorder" cannot be a constant, if not then it should be refactored.